### PR TITLE
VisibilityManipulator adjustment

### PR DESCRIPTION
**Pull request by Bl00D4NGEL**
Previously the VisibilityManipulator would strip public, protected and/or private visibility. This caused the ChangeMethodVisibilityRector to cause problems described in https://github.com/rectorphp/rector/issues/6829

refs: https://github.com/rectorphp/rector/issues/6829
closes: https://github.com/rectorphp/rector/issues/6829

### CODE REVIEW

**Comment by Bl00D4NGEL** [comment]
@TomasVotruba please review. My changes don't seem to be the cause for the rector-nette step to fail. :)

File: rules/Privatization/NodeManipulator/VisibilityManipulator.php [hunk 1]
```diff
@@ -157,7 +157,7 @@ private function replaceVisibilityFlag(ClassMethod | Property | ClassConst $node
         $this->addVisibilityFlag($node, $visibility);
 
         if ($isStatic) {
-            $this->makeStatic($node);
+            $this->addVisibilityFlag($node, Visibility::STATIC);
```
Comment (TomasVotruba) [inline rules/Privatization/NodeManipulator/VisibilityManipulator.php]: Why this change? Now we have to maintain the same code on 2 places, which might lead to change of one without another.

File: rules/Privatization/NodeManipulator/VisibilityManipulator.php [hunk 2]
```diff
@@ -147,7 +147,7 @@ private function replaceVisibilityFlag(ClassMethod | Property | ClassConst $node
     {
         $isStatic = $node instanceof ClassMethod && $node->isStatic();
         if ($isStatic) {
-            $this->removeVisibility($node);
+            $this->removeVisibilityFlag($node, Visibility::STATIC);
```
Comment (TomasVotruba) [inline rules/Privatization/NodeManipulator/VisibilityManipulator.php]: :+1:

File: rules-tests/Privatization/NodeManipulator/VisibilityManipulatorTest.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+declare(strict_types=1);
+
+namespace Rector\Tests\Privatization\NodeManipulator;
+
+use PhpParser\Node\Stmt\ClassMethod;
+use Rector\Core\ValueObject\Visibility;
+use Rector\Privatization\NodeManipulator\VisibilityManipulator;
+use Rector\Testing\PHPUnit\AbstractTestCase;
+
+final class VisibilityManipulatorTest extends AbstractTestCase
```
Comment (TomasVotruba) [inline rules-tests/Privatization/NodeManipulator/VisibilityManipulatorTest.php:12]: :+1:

**Comment by TomasVotruba** [comment]
Thank you for the test and fix :clap: amazing work.

I'll check why the Nette package fails, last PRs today it was working fine.

**Comment by TomasVotruba** [comment]
Could you rebase on current `main`? I tried to cleanup package testing CI setup

File: rules/Privatization/NodeManipulator/VisibilityManipulator.php (hunk 1 above)
Comment (Bl00D4NGEL) [reply rules/Privatization/NodeManipulator/VisibilityManipulator.php]: I just tried to be as close to the previous code as possible without breaking other functionality.
We need to apparently strip the static tag so the visibility is correctly adjusted. (If I would not strip it other tests are failing)
What do you exactly mean by "maintain the same coe on 2 places"? Maybe I can find a way to improve this.

Comment (TomasVotruba) [reply rules/Privatization/NodeManipulator/VisibilityManipulator.php]: The code seems 100 % same as in `makeStatic()`.

> What do you exactly mean by "maintain the same coe on 2 places"? Maybe I can find a way to improve this.

If there is bug in `makeStatic()`, now we have to fix in 2 places.
How does this fix the reported bug?

Comment (Bl00D4NGEL) [reply rules/Privatization/NodeManipulator/VisibilityManipulator.php]: I honestly copied from makeStatic and forgot that I did that when trying to look for the solution 😅 Adjusted the calls to use the makeStatic and makeNonStatic functions instead.

Previously we were removing the visibility modifier (which I don't quite understand why) because it somehow fixed https://github.com/rectorphp/rector/issues/4941 in https://github.com/rectorphp/rector/pull/4951 but it looks like this change is actually the only one required to fix the problem listed in rectorphp/rector#6829.
I didn't dig deep enough to find out why this is the case though.

**Review by samsonasik — APPROVED** [review]

**Review by TomasVotruba — APPROVED** [review]

**Comment by TomasVotruba** [comment]
Thank you, this looks solid 🤗
